File: 04-factorial/factorial.rb
# Ruby program to compute factorial
# Created by M.V.Harish Kumar on 10-09-2023
<<<<<<< HEAD
class Integer
    def !
        (1..self).inject(:*)
	  end
=======

def factorial(n)
    (1..n).inject(:*)
>>>>>>> 1bc10791fb251e245e935b4186828b216d245e39
end

print "Enter value for `n`: "
n = gets.to_i
<<<<<<< HEAD
puts "#{n}! = #{!n}"
=======
puts "#{n}! = #{factorial n}"
>>>>>>> 1bc10791fb251e245e935b4186828b216d245e39

File: 04-factorial/factorial.php
<?php
// PHP program to compute factorial
// Created by M.V.Harish Kumar on 17-09-2024
function factorial($n) {
	return $n <= 1 ? 1 : $n*factorial($n-1);
}

$n = intval(readline("Enter value for `n`: "));
echo "$n! = " . factorial($n) . chr(10);
